### DISCUSSION TITLE
Customize logo-symbol-wordmark.svg in web admin or instruction in docs.joinmastodon.org

### DISCUSSION THREAD

**Discussion by Jim-Young** [mastodon/mastodon#27878] [General]
### Pitch

As a server admin I'd like to change the logo similar to https://social.vivaldi.net/

![Selection_001](https://github.com/mastodon/mastodon/assets/29013899/8ad591e6-867d-4969-85de-0b78797f0585)

I can see the svg file is in /app/javascript/images/logo-symbol-wordmark.svg, but whenever I replace it with my customized logo, it doesn't show on the page, like empty area.
![Selection_003](https://github.com/mastodon/mastodon/assets/29013899/f6e8b730-7514-4658-a4f7-14283979fe8a)

So far I've tried my own svg files and even downloaded the svg from Vivaldi Social, it still the same.

I've also tried:

systemctl restart mastodon-web mastodon-sidekiq mastodon-streaming
RAILS_ENV=production bundle exec rails assets:precompile after copying the svg to the server
restart the server

I'm out of idea as not a rail person.

Can anyone help please?

### Motivation

Different logos can distinguish site to site and make fediverse in real. We are different, we want different looking.

**Comment by Jim-Young** [comment]
OK, I figured out why svg not accepted as wordmark logo.

In the svg file, there must be a symbol id which used by react

in case someone has same trouble, at least you can manually change the logo

`<symbol id="logo-symbol-wordmark">
</symbol> `

**Comment by Nikoh77** [comment]
> OK, I figured out why svg not accepted as wordmark logo.
> 
> In the svg file, there must be a symbol id which used by react
> 
> in case someone has same trouble, at least you can manually change the logo
> 
> `<symbol id="logo-symbol-wordmark"> </symbol> `

Hello, please can you explain me better how i should do it?

**Comment by Jim-Young** [comment]
> OK, I figured out why svg not accepted as wordmark logo.
> 
> In the svg file, there must be a symbol id which used by react
> 
> in case someone has same trouble, at least you can manually change the logo
> 
> `<symbol id="logo-symbol-wordmark"> </symbol> `

First to replace the logo-symbol-wordmark.svg with yours in /home/mastodon/live/app/javascript/images

then, manually edit logo-symbol-wordmark.svg or edit it before you replace this file. Make sure your paths are inside <use
`<use xlink:href="#logo-symbol-wordmark"/>
<symbol id="logo-symbol-wordmark">`

![Selection_005](https://github.com/mastodon/mastodon/assets/29013899/1ced0ae1-73e5-4dbe-9810-9e85ce63c874)
![Selection_006](https://github.com/mastodon/mastodon/assets/29013899/c5295492-72d2-4f8c-98db-1544236fa656)

It should work now.

Alternatively, if your 

`<use xlink:href="#logo-symbol-wordmark"/>
<symbol id="logo-symbol-wordmark">`

are different, you can edit logo.js

`vi /home/mastodon/live/app/javascript/mastodon/components/logo.js`

After than, you will need to precompile and reload web

`RAILS_ENV=production bundle exec rails assets:precompile`

`systemctl reload mastodon-web`

**Comment by mrAceT** [comment]
did something change with 4.1.4.. what is written above doesn't seem to work anymore!?

**Comment by alpacaccp** [comment]
> did something change with 4.1.4.. what is written above doesn't seem to work anymore!?

It still works.

**Comment by mrAceT** [comment]
> > did something change with 4.1.4.. what is written above doesn't seem to work anymore!?
> It still works.

Thanks, then something changed at my end..

**Comment by nickjuntilla** [comment]
What directory do you run RAILS_ENV=production bundle exec rails assets:precompile from? Do you need to install something called "bundle"? I get Command 'bundle' not found. I read an article that said the bundle command is deprecated. Are people still using this command?

**Comment by Jim-Young** [comment]
> What directory do you run RAILS_ENV=production bundle exec rails assets:precompile from? Do you need to install something called "bundle"? I get Command 'bundle' not found. I read an article that said the bundle command is deprecated. Are people still using this command?

Not sure where you get the bundle command deprecated info.
The default directory should be /home/mastodon if you followed the official installation guide and you run bundle from there.
